### PR TITLE
Improve error for invalid ranges

### PR DESCRIPTION
**Pull request by westonganger**
Improve error messages when incorrect ranges are provided to `Worksheet#[]`
  - Related to https://github.com/axlsx-styler-gem/axlsx_styler/issues/33 and the other referenced axlsx issues

### CODE REVIEW

File: lib/axlsx/workbook/worksheet/worksheet.rb [hunk 1]
```diff
@@ -611,9 +619,13 @@ def [](cell_def)
     # @param [String] name The cell or cell range to return. "A1" will return the first cell of the first row.
     # @return [Cell]
     def name_to_cell(name)
-      col_index, row_index = *Axlsx::name_to_indices(name)
+      row_index, col_index = *Axlsx::name_to_indices(name)
```
Comment (westonganger) [inline lib/axlsx/workbook/worksheet/worksheet.rb:622]: This line is the fix for Item B

File: lib/axlsx/workbook/worksheet/worksheet.rb [hunk 2]
```diff
@@ -599,10 +599,18 @@ def relationships
     # @return [Cell, Array]
     def [](cell_def)
       return rows[cell_def] if cell_def.is_a?(Integer)
+
       parts = cell_def.split(':').map{ |part| name_to_cell part }
+
       if parts.size == 1
         parts.first
       else
+        if parts.size > 2
```
Comment (westonganger) [inline lib/axlsx/workbook/worksheet/worksheet.rb:608]: Here is the improvements for Item A

File: lib/axlsx.rb [hunk 3]
```diff
@@ -79,11 +79,25 @@ def self.coder
   # returns the x, y position of a cell
   def self.name_to_indices(name)
```
Comment (westonganger) [inline lib/axlsx.rb:80]: While debugging this method I simply refractored this method to be much more understandable. So no actual changes here.

Comment (straydogstudio) [reply lib/axlsx.rb:80]: It is certainly less cryptic. I think it can stay.

**Comment by straydogstudio** [comment]
@simi @noniq Any comments on this PR

**Comment by straydogstudio** [comment]
@kiskoza You too. Any comments

**Comment by straydogstudio** [comment]
@westonganger In general we put our error messages in lib/axlsx/util/constants.rb. Dynamic error messages usually have %s for use with printf or the like. Would you change the new error messages so they aren't hard coded?

FYI @simi @noniq @kiskoza

**Comment by kiskoza** [comment]
The code looks good to me other than that small change request @straydogstudio asked for

**Comment by westonganger** [comment]
Ok I have updated with the requested changes

**Pull request merged**